### PR TITLE
Refactor and update to C# 7 constructs

### PR DESCRIPTION
**Pull request by BillWagner**
Controllers/BasketController:
Use Expression Bodied Members.
Delete() should not be 'fire and forget'

Controllers/HomeController:
Expression bodied constructor

IntegrationEvents/EventHandling/ProductPriceChangedIntegratedEventHandler:
Use valueTask to minimize allocations for 'fast path' tasks. One might
believe that
most users would not have the changing item in the basket at any given
time.
Using ValueTask instead of Task, this minimizes allocations to only when
async
work must actually take place.

IntegrationEvents/EventHandling/ProductPriceChangedIntegratedEvent:
Make the properties ReadOnly

Model/Basket:
Use property initializer

Model/RedisBasketRepository:
Remove redundant ToString()
Use elvis operator instead of statement null check

Startup:
remove redundant ToString()

### CODE REVIEW

File: src/Services/Basket/Basket.API/IntegrationEvents/EventHandling/ProductPriceChangedIntegrationEventHandler.cs [hunk 1]
```diff
@@ -5,6 +5,7 @@
 using System.Linq;
 using System.Threading.Tasks;
 
+
```
Comment (jzeferino) [inline src/Services/Basket/Basket.API/IntegrationEvents/EventHandling/ProductPriceChangedIntegrationEventHandler.cs]: Empty line should be removed.

File: src/Services/Basket/Basket.API/IntegrationEvents/EventHandling/ProductPriceChangedIntegrationEventHandler.cs [hunk 2]
```diff
@@ -16,7 +17,7 @@ public ProductPriceChangedIntegrationEventHandler(IBasketRepository repository)
             _repository = repository ?? throw new ArgumentNullException(nameof(repository));
         }
 
-        public async Task Handle(ProductPriceChangedIntegrationEvent @event)
+         public async Task Handle(ProductPriceChangedIntegrationEvent @event)
```
Comment (jzeferino) [inline src/Services/Basket/Basket.API/IntegrationEvents/EventHandling/ProductPriceChangedIntegrationEventHandler.cs]: Space before method should be removed.

File: src/Services/Basket/Basket.API/IntegrationEvents/EventHandling/ProductPriceChangedIntegrationEventHandler.cs [hunk 3]
```diff
@@ -32,7 +33,7 @@ public ProductPriceChangedIntegrationEventHandler(IBasketRepository repository)
         {
             var itemsToUpdate = basket?.Items?.Where(x => int.Parse(x.ProductId) == productId).ToList();
 
-            if (itemsToUpdate != null)
+            if ((itemsToUpdate != null) && (itemsToUpdate.Any()))
```
Comment (jzeferino) [inline src/Services/Basket/Basket.API/IntegrationEvents/EventHandling/ProductPriceChangedIntegrationEventHandler.cs]: Could be simplified to `itemsToUpdate?.Any()`

**Comment by eiximenis** [comment]
Closing as is too old, and much of this has been already done :)

**Pull request closed without merging**